extensions/ql-vscode/src/model-editor/method-modeling/method-modeling-view-provider.ts

```diff
@@ -46,14 +46,17 @@ export class MethodModelingViewProvider extends AbstractWebviewViewProvider<
   }
 
   private setInitialState(): void {
-    const selectedMethod = this.modelingStore.getSelectedMethodDetails();
-    if (selectedMethod) {
-      void this.postMessage({
-        t: "setSelectedMethod",
-        method: selectedMethod.method,
-        modeledMethod: selectedMethod.modeledMethod,
-        isModified: selectedMethod.isModified,
-      });
+    const activeState = this.modelingStore.getStateForActiveDb();
+    if (activeState) {
+      const selectedMethod = this.modelingStore.getSelectedMethodDetails();
```

Comment:
It seems like we're only retrieving the `activeDb` to check whether there is any state, and `getSelectedMethodDetails` will access the state again inside the modeling store. This seems somewhat redundant, but throwing an error inside `getSelectedMethodDetails` does seem very useful. Could we perhaps add a method/getter in the modeling store like `hasStateForActiveDb` to make the intent more clear?